internal/cmd/deployorder.go

```diff
@@ -0,0 +1,128 @@
+package cmd
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/simplesurance/dependencies-tool/internal/deps"
+)
+
+const deployOrderShortHelp = "Generate a deployment order."
+
+var descrDependencyFileNames = strings.TrimSpace(`
+Dependency definition files are discovered by searching in all child directories
+of PATH.
+Files that match the following names are discovered, only the first found per
+directory is parsed, the preference order is:
+
+  1. .deps-<ENVIRONMENT>-<REGION>.toml
+  2. .deps-<ENVIRONMENT>.toml
+  2. .deps.toml
+`)
+
+var deployOrderLongHelp = deployOrderShortHelp + "\n\n" + strings.TrimSpace(`
+Positional Arguments:
+  PATH		- Root Directory for the dependency file discovery.
+  ENVIRONMENT   - Value that is used as the ENVIRONMENT placeholder of the 
+                  searched dependency file names.
+  REGION        - Value that is used as the REGION placeholder of the searched
+		  dependency file names.
+  APP-NAME      - Application names for that the deployment order is generated.
+                  If not specified, the dependencies of all applications are 
+		  evaluated.
```

Comment:
Instead of having an `[APP-NAME]...` positional argument, we could also have a "--filter" parameter accepting a comma-separated list of strings.
By default it would be empty, therefore dependencies of all applications would be used.
We currently also support the "not: <APP-NAME>" syntax, this fits the`--filter` term well.

I'm undecided if that would be better then having it as positional argument, any opinions?

Reply:
why do we need to support the `not:` syntax?

Reply:
If will not support `not:` in the future, the parameter could be named `--apps`  instead of `--filter`.

An alternative could be to make APP-NAME a required parameter, so that "all" has to be passed explicitly instead of assuming it as default.

Reply:
> Good question, our deploy Jenkins job still supports and documents it. I don't know if anybody uses it.
> 
> I'll check if we can get rid of it.

I was wrong.
We actually don't support in anymore for sisu.git deployments since July and have it documented.
I'll remove support for it in another PR in the  dependencies-tool.

Reply:
I'll make apps a named parameter.
It allows easier validation and the usage description also becomes a bit easier when support is added to read the  generated dependency-tree from a file.

Reply:
done, I also updated the usage output in the PR description